dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -496,3 +507,96 @@ fn load_manifest(path: &Path) -> Result<DscResource, DscError> {
 
     Ok(resource)
 }
+
+fn sort_adapters_based_on_lookup_table(unsorted_adapters: &BTreeMap<String, Vec<DscResource>>, needed_resource_types: &Vec<String>) -> LinkedHashMap<String, Vec<DscResource>>
+{
+    let mut result:LinkedHashMap<String, Vec<DscResource>> = LinkedHashMap::new();
+    let lookup_table:HashMap<String, String> = load_adapted_resources_lookup_table();
+    // first add adapters (for needed types) that can be found in the lookup table
+    for needed_resource in needed_resource_types {
+        if let Some(adapter_name) = lookup_table.get(needed_resource) {
+            if let Some(resource_vec) = unsorted_adapters.get(adapter_name) {
+                debug!("Lookup table found resource '{}' in adapter '{}'", needed_resource, adapter_name);
+                result.insert(adapter_name.to_string(), resource_vec.clone());
+            }
+        }
+    }
+
+    // now add remaining adapters
+    for (adapter_name, adapters) in unsorted_adapters {
+        if !result.contains_key(adapter_name) {
+            result.insert(adapter_name.to_string(), adapters.clone());
+        }
+    }
+
+    result
+}
+
+fn add_resources_to_lookup_table(adapted_resources: &BTreeMap<String, Vec<DscResource>>)
+{
+    let mut lookup_table:HashMap<String, String> = load_adapted_resources_lookup_table();
+
+    for (resource_name, res_vec) in adapted_resources {
+        let adapter_name = &res_vec[0].require_adapter.as_ref().unwrap();
+        lookup_table.insert(resource_name.to_string().to_lowercase(), (*adapter_name).to_string());
+    };
+
+    save_adapted_resources_lookup_table(&lookup_table);
+}
+
+fn save_adapted_resources_lookup_table(lookup_table: &HashMap<String, String>)
+{
+    if let Ok(lookup_table_json) = serde_json::to_string(&lookup_table) {
+        let file_path = get_lookup_table_file_path();
+        debug!("Saving lookup table with {} items to {:?}", lookup_table.len(), file_path);
+
+        let path = std::path::Path::new(&file_path);
+        if let Some(prefix) = path.parent() {
+            if fs::create_dir_all(prefix).is_ok()  {
+                if fs::write(file_path.clone(), lookup_table_json).is_err() {
+                    debug!("Unable to write lookup_table file {file_path:?}");
+                }
+            } else {
+                debug!("Unable to create parent directories of the lookup_table file {file_path:?}");
+            }
+        } else {
+            debug!("Unable to get directory of the lookup_table file {file_path:?}");
+        }
+    } else {
+        debug!("Unable to serialize lookup_table to json");
+    }
+}
+
+fn load_adapted_resources_lookup_table() -> HashMap<String, String>
+{
+    let file_path = get_lookup_table_file_path();
+    
+    let lookup_table: HashMap<String, String> = match fs::read(file_path.clone()){
```

Comment:
looks like `fs::read()` takes a ref, so you don't need to clone?

Reply:
Without `clone` the debug statement later on (line 586) throws `value borrowed here after move`.